lib/improver/blending/weighted_blend.py

```diff
@@ -486,6 +484,346 @@ def __repr__(self):
                        ' coord_adjust = {2}>')
         return description.format(self.coord, self.mode, self.coord_adjust)
 
+    def is_cube_percentile_data(self, cube):
+        """
+        Determines if the cube to be blended has a percentile dimension
+        coordinate.
+
+        Args:
+            cube (iris.cube.Cube):
+                The cube to be checked for a percentile coordinate.
+        Returns:
+            None or perc_coord (iris.coords.DimCoord):
+                None if no percentile dimension coordinate is found. If
+                such a coordinate is found it is returned.
+        Raises:
+            ValueError : If there is a percentile coord and it is not a
+                dimension coord in the cube.
+            ValueError : If there is a percentile dimension with only one
+                point, we need at least two points in order to do the blending.
+            ValueError : If there is a percentile dimension on the cube and the
+                mode for blending is 'weighted_maximum'
+        """
+        try:
+            perc_coord = find_percentile_coordinate(cube)
+            perc_dim = cube.coord_dims(perc_coord.name())
+            if not perc_dim:
+                msg = ('The percentile coord must be a dimension '
+                       'of the cube.')
+                raise ValueError(msg)
+            # Check the percentile coordinate has more than one point,
+            # otherwise raise an error as we won't be able to blend.
+            if len(perc_coord.points) < 2.0:
+                msg = ('Percentile coordinate does not have enough points'
+                       ' in order to blend. Must have at least 2 percentiles.')
+                raise ValueError(msg)
+            # If we have a percentile dimension and the mode is 'max' raise an
+            # exception.
+            if perc_coord and self.mode == 'weighted_maximum':
+                msg = ('The "weighted_maximum" mode cannot be used with'
+                       ' percentile data.')
+                raise ValueError(msg)
+
+            return perc_coord
+        except CoordinateNotFoundError:
+            return None
+
+    def check_compatible_time_points(self, cube):
+        """
+        Check that the time coordinate only contains a single time if the
+        coordinate being blended is forecast_reference_time.
```

Comment:
I have simplified the implementation of a test that claimed to be doing more than it was. We should discuss what you think it needs to do in reality; I'm not sure I know what you would like.

Reply:
I rather think it should check there is only one time point regardless of the blend coordinate.  But if there's any question over that we should probably put it as a follow on ticket in "technical nice-to-haves".

Reply:
As discussed: yes, this would be better as a separate PR please, but don't spend too much time on it as I'll have to do this work anyway as part of the "integer seconds in unit tests" change.

Reply:
See PR #739